emt4j-common/src/main/java/org/eclipse/emt4j/common/rule/impl/WholeClassRule.java

```diff
@@ -58,7 +61,15 @@ protected CheckResult check(Dependency dependency) {
         mvelMap.put("cpSet", dependency.getClassSymbol().getConstantPoolSet());
         Object result = MVEL.eval(mvel2Rule, mvelMap);
         if (result instanceof Boolean) {
-            return ((Boolean) result) ? CheckResult.FAIL : CheckResult.PASS;
+            if (staticAnalysisRule == null) {
+                return ((Boolean) result) ? CheckResult.FAIL : CheckResult.PASS;
+            } else {
+                if ((Boolean) result) {
+                    boolean found = StaticAnalysisEntry.analyze(staticAnalysisRule, dependency.getTarget().asClass().getClassName(), dependency.getCurrClassBytecode());
+                    return found ? CheckResult.FAIL : CheckResult.PASS;
```

Comment:
The analysis implemented here is simple, and not sound. The result yes is true positive, but a no may be false negative. Didn't see how you handle the false negatives.

Reply:
Yes, but I think this is the most thing that the current implementation(the scope of static analysis is a sole method rather than inter methods) can do.
We can extend the ability in other patches.